Drop-In/src/androidTest/java/com/braintreepayments/api/SelectPaymentMethodFragmentUITest.kt

```diff
@@ -0,0 +1,236 @@
+package com.braintreepayments.api
+
+import androidx.core.os.bundleOf
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.lifecycle.Lifecycle
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.braintreepayments.api.dropin.R
+import org.hamcrest.Matchers.not
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+
+// Ref: https://developer.android.com/guide/fragments/test
+@RunWith(AndroidJUnit4::class)
+class SelectPaymentMethodFragmentUITest {
+
+    private lateinit var countDownLatch: CountDownLatch
+
+    private val supportedPaymentMethods = listOf(
+            DropInPaymentMethodType.PAYPAL,
+            DropInPaymentMethodType.PAY_WITH_VENMO,
+            DropInPaymentMethodType.UNKNOWN,
+            DropInPaymentMethodType.GOOGLE_PAYMENT
+    )
+
+    @Before
+    fun beforeEach() {
+        countDownLatch = CountDownLatch(1)
+    }
+
+    @Test
+    @Throws(InterruptedException::class)
+    fun whenStateIsRESUMED_loaderIsVisible() {
+        val scenario = FragmentScenario.launchInContainer(SelectPaymentMethodFragment::class.java)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+        onView(withId(R.id.bt_select_payment_method_loader)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenSupportedPaymentMethodsLoaded_displaysPaymentMethods() {
+        val scenario = FragmentScenario.launchInContainer(SelectPaymentMethodFragment::class.java)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedPaymentMethods(supportedPaymentMethods)
+        }
+
+        onView(isRoot()).perform(waitFor(500))
+        onView(withId(R.id.bt_select_payment_method_loader)).check(matches(not(isDisplayed())))
+        onView(withText("PayPal")).check(matches(isDisplayed()))
+        onView(withText("Venmo")).check(matches(isDisplayed()))
+        onView(withText("Credit or Debit Card")).check(matches(isDisplayed()))
+        onView(withText("Google Pay")).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenSupportedPaymentMethodsLoaded_requestsForVaultedPaymentMethodsToBeLoaded() {
+        // TODO: assert an event is dispatched to load vaulted payment methods
```

Comment:
Is this waiting for a future PR or can it be tested now?

Reply:
We may punt on unit testing analytics in the short term. Since multiple analytics can be thrown for a given set of UI interactions, we need a repeatable way to assert that a specific analytics event is emitted by the fragment.

Reply:
We have some ideas but the bulk of this PR is about the fragment refactor. It'll be super boilerplate.